File: module02/src/ss7_abstractclass_interface/geometric_resizeable/Rectangle.java
package ss7_abstractclass_interface.geometric_resizeable;

public class Rectangle extends Shape implements Resizeable {
    private double width = 10.0;
    private double height = 12.0;

    public Rectangle() {

    }

    public Rectangle(double width, double height) {
        this.width = width;
        this.height = height;
    }

    public Rectangle(String color, boolean filled, double width, double height) {
        super(color, filled);
        this.width = width;
        this.height = height;
    }

    public double getWidth() {
        return width;
    }

    public void setWidth(double width) {
        this.width = width;
    }

    public double getHeight() {
        return height;
    }

    public void setHeight(double height) {
        this.height = height;
    }

    public double getArea() {
        return height * width;
    }

    public double getPerimeter() {
        return (height + width) * 2;
    }

    @Override
    public String toString() {
        return "Rectangle{" +
                "width=" + getWidth() +
                ", height=" + getHeight() +
                '}';
    }

    @Override
    public double resize(double percent) {
        return getArea() * (100 * percent) / 100;
    }
}

File: module05/ss02_typescript/sharp/src/rectangle.ts
import { Sharp } from "./Sharp";
import { IArea } from "./IArea";

export class Rectangle extends Sharp implements IArea{

  constructor(x: number, y: number,private _width:number,private _height:number) {
    super(x, y);
  }

  area(): number {
    return  this.x*this.y;
  }

  get width() {
    return this._width;
  }

  set width(value) {
    this._width = value;
  }

  get height() {
    return this._height;
  }

  set height(value) {
    this._height = value;
  }

 toString(): string {
    return "width: " +this.x +
            " height: "+this.y ;
  }
}
